Firestore/Source/Core/FSTSyncEngine.mm

```diff
@@ -344,15 +342,16 @@ - (void)applyChangedOnlineState:(FSTOnlineState)onlineState {
   [self.delegate handleViewSnapshots:newViewSnapshots];
 }
 
-- (void)rejectListenWithTargetID:(FSTBoxedTargetID *)targetID error:(NSError *)error {
+- (void)rejectListenWithTargetID:(const TargetId)targetID error:(NSError *)error {
   [self assertDelegateExistsForSelector:_cmd];
 
-  FSTDocumentKey *limboKey = self.limboKeysByTarget[targetID];
-  if (limboKey) {
+  const auto iter = _limboKeysByTarget.find(targetID);
+  if (iter != _limboKeysByTarget.end()) {
+    const DocumentKey limboKey = iter->second;
```

Comment:
Very optional: this can be a const reference.

Reply:
Actually we want a copy since the line below `_limboKeysByTarget.erase(targetID);` and later need to access the `limboKey`.

Reply:
Ah, sorry, I missed the usage below the diff.